core/store/src/trie/config.rs

```diff
@@ -111,10 +102,13 @@ impl ShardCacheConfig {
     }
 
     fn total_size_limit(&self, shard_uid: ShardUId) -> u64 {
-        self.override_max_total_bytes
+        let explicit_limit = self
+            .override_max_total_bytes
             .get(&shard_uid)
             .cloned()
-            .unwrap_or(self.default_max_total_bytes)
+            .unwrap_or(self.default_max_total_bytes);
+        let implicit_limit = self.capacity(shard_uid) * TrieCacheInner::PER_ENTRY_OVERHEAD;
+        explicit_limit.min(implicit_limit)
```

Comment:
could you add a comment explaining this logic? Ie, why not just one limit? 

Also, would we look into simplifying this in the future?

Reply:
Added sine comments, I hope they answer the question why we need it.
And yes, this is very much a temporary solution that we want to remove a few releases after #7578, which deprecates per entry config.

Reply:
important note: changed the formula for the implicit limit, I believe it was wrong before, restricting it too much
Now we allow the old per-entry config to grow larger than before in non-worst cases. That makes much more sense to me than making cache capacity smaller in all but the case where all values are empty.

Reply:
I linked TODOs to this issue: https://github.com/near/nearcore/issues/7894

For marking it as deprecated, it would be too early in this PR. There is a separate PR #7578 (that I will update today) that deals with the deprecation issues.